### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by alittlebitweird**
* refactor - rewrite of editframe-js
* type-safety - add type-guards for incoming api data
* validation - validate user input at runtime in addition to ts declarations
* tests - test all the things ~93% coverage
* modular - files split into api/constant/features/mocks/strings/utils
* cleanup - remove unused packages, files, comments / auto-docs
* prettier - autoformatting on save
* linting - in-depth eslint configuration to maintain consistency
* pre-commit - runs autoformatting, ts error-checking, linting, tests

### CODE REVIEW

File: src/utils/strings/strings.test.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { uuid } from './'
+
+describe('uuid', () => {
+  it('genreates a 6-digit uuid', () => {
```
Comment (brycedev) [inline src/utils/strings/strings.test.ts:4]: small typo

File: src/features/videos/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        validate: isVideo,
+      })
+    } catch (error) {
+      console.error(VideoErrorText.get(error.message))
+    }
+
+    return undefined
+  }
+
+  public [ApiVideoMethod.new](options: CompositionOptions): Composition {
```
Comment (brycedev) [inline src/features/videos/index.ts:46]: how do you feel about using .create vs .new since the latter is a reserved keyword?

Comment (alittlebitweird) [reply src/features/videos/index.ts:46]: Love it

Comment (alittlebitweird) [reply src/features/videos/index.ts:46]: Actually now I remember why I chose new. Becsuse it's not actually completing a creation, it's just instantiating it and waiting for you to call encode. Thoughts?

Comment (brycedev) [reply src/features/videos/index.ts:46]: yeah, i definitely follow that. we can keep .new unless something arises, then

**Comment by brycedev** [comment]
approved

**Pull request merged**